src/Components/CreateImageWizardV2/steps/TargetEnvironment/Azure/AzureSourcesSelect.tsx

```diff
@@ -1,28 +1,36 @@
 import React, { useState, useEffect } from 'react';
 
-import FormSpy from '@data-driven-forms/react-form-renderer/form-spy';
-import useFieldApi from '@data-driven-forms/react-form-renderer/use-field-api';
-import useFormApi from '@data-driven-forms/react-form-renderer/use-form-api';
 import { Alert } from '@patternfly/react-core';
 import { FormGroup, Spinner } from '@patternfly/react-core';
 import {
   Select,
   SelectOption,
   SelectVariant,
 } from '@patternfly/react-core/deprecated';
-import PropTypes from 'prop-types';
 
-import { extractProvisioningList } from '../../../store/helpers';
+import { useAppDispatch, useAppSelector } from '../../../../../store/hooks';
 import {
   useGetSourceListQuery,
   useGetSourceUploadInfoQuery,
-} from '../../../store/provisioningApi';
+} from '../../../../../store/provisioningApi';
+import {
+  changeAzureResourceGroup,
+  changeAzureSource,
+  changeAzureSubscriptionId,
+  changeAzureTenantId,
+  selectAzureSource,
+  selectAzureSubscriptionId,
+  selectAzureTenantId,
+} from '../../../../../store/wizardSlice';
 
-const AzureSourcesSelect = ({ label, isRequired, className, ...props }) => {
-  const { change } = useFormApi();
-  const { input } = useFieldApi(props);
+export const AzureSourcesSelect = () => {
+  let tenantId = useAppSelector((state) => selectAzureTenantId(state));
```

Comment:
This should be defined using `const`, not `let`. The `useAppSelector` hook creates a subscription - see the docs:

https://react-redux.js.org/api/hooks
>useSelector() will also subscribe to the Redux store, and run your selector whenever an action is dispatched.

If we re-assign the value later as done here, we lose the benefit of having a subscription.